### PR TITLE
[EmbeddingApi-usecase] Add usecase for API ClearCacheForSingleFile to…

### PR DESCRIPTION
**Pull request by jacky-young**
… clear cache for single file test

-Add usecase for API ClearCacheForSingleFile to clear cache for single file test
-Cover the XWalkActivity and XWalkInitializer two ways

Impacted tests(approved): new 2, update 0, delete 0
Unit test platform: [Android]
Unit test result summary: pass 2, fail 0, block 0

https://crosswalk-project.org/jira/browse/XWALK-4609

### CODE REVIEW

**Comment by jacky-young** [comment]
@jennycao please help to review it.

**Comment by jennycao** [comment]
@Shao-Feng Pls help to review

File: usecase/usecase-embedding-android-tests/embeddingapi-async/src/org/xwalk/embedded/api/asyncsample/XWalkViewWithClearCacheForSingleFileAsync.java [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.view.View.OnClickListener;
+import android.webkit.WebResourceResponse;
+import android.widget.Button;
+import android.widget.TextView;
+
+public class XWalkViewWithClearCacheForSingleFileAsync extends Activity implements XWalkInitializer.XWalkInitListener {
+	private static final String TAG = "XWalkViewWithClearCacheForSingleFileAsync";
```
Comment (Shao-Feng) [inline usecase/usecase-embedding-android-tests/embeddingapi-async/src/org/xwalk/embedded/api/asyncsample/XWalkViewWithClearCacheForSingleFileAsync.java]: The TAG string is too long. a warnning will be raised once using this TAG to print Log

File: usecase/usecase-embedding-android-tests/embeddingapi-async/src/org/xwalk/embedded/api/asyncsample/XWalkViewWithClearCacheForSingleFileAsync.java [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+	private static final String EXTRA_CACHE_PATH
+				= "/data/data/org.xwalk.embedded.api.sample/app_xwalkcore/Default/Cache";
```
Comment (Shao-Feng) [inline usecase/usecase-embedding-android-tests/embeddingapi-async/src/org/xwalk/embedded/api/asyncsample/XWalkViewWithClearCacheForSingleFileAsync.java]: Do we really need to hard code the cache path? if the package name is changed, the path will need to be updated too.
How about use the Activity.getCacheDir()?

File: usecase/usecase-embedding-android-tests/embeddingapi-async/src/org/xwalk/embedded/api/asyncsample/XWalkViewWithClearCacheForSingleFileAsync.java [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+		}
+    	return FormetFileSize(blockSize);
+    }
+    
+    private static long getFileSize(File f) throws Exception{
+		// TODO Auto-generated method stub
+    	long size = 0;
+    	if (f.exists()) {
+    	     FileInputStream fis = new FileInputStream(f);
+    	     size = fis.available();
```
Comment (Shao-Feng) [inline usecase/usecase-embedding-android-tests/embeddingapi-async/src/org/xwalk/embedded/api/asyncsample/XWalkViewWithClearCacheForSingleFileAsync.java]: http://stackoverflow.com/questions/116574/java-get-file-size-efficiently
stream.available() does not return the file length. It returns the amount of bytes which are available for read without blocking other streams. It is not necessarily the same amount of bytes as file length. To get the real length from a stream, you really need to read it (and count the read bytes meanwhile)

How about use the File or URL method to get the filesize

File: usecase/usecase-embedding-android-tests/embeddingapi-async/src/org/xwalk/embedded/api/asyncsample/XWalkViewWithClearCacheForSingleFileAsync.java (hunk 2 above)
Comment (jacky-young) [inline usecase/usecase-embedding-android-tests/embeddingapi-async/src/org/xwalk/embedded/api/asyncsample/XWalkViewWithClearCacheForSingleFileAsync.java]: getCacheDir() can only get data/data//cache, this specific path is checked with hengzhi, it is just about xwalk_core as it shows.

File: usecase/usecase-embedding-android-tests/embeddingapi-async/src/org/xwalk/embedded/api/asyncsample/XWalkViewWithClearCacheForSingleFileAsync.java [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+				size = size + getFileSizes(flist[i]);
+			}
+			else{
+				size = size + getFileSize(flist[i]);
+			}
+		}
+		return size;
+	}
+	
+	private static String FormetFileSize(long fileS) {
```
Comment (Shao-Feng) [inline usecase/usecase-embedding-android-tests/embeddingapi-async/src/org/xwalk/embedded/api/asyncsample/XWalkViewWithClearCacheForSingleFileAsync.java]: http://stackoverflow.com/questions/3263892/format-file-size-as-mb-gb-etc

please refer

File: usecase/usecase-embedding-android-tests/embeddingapi-async/src/org/xwalk/embedded/api/asyncsample/XWalkViewWithClearCacheForSingleFileAsync.java (hunk 1 above)
Comment (jacky-young) [inline usecase/usecase-embedding-android-tests/embeddingapi-async/src/org/xwalk/embedded/api/asyncsample/XWalkViewWithClearCacheForSingleFileAsync.java]: How about TAG = "XWalk_ClearCacheForSingleFile"?

File: usecase/usecase-embedding-android-tests/embeddingapi/src/org/xwalk/embedded/api/sample/XWalkViewWithClearCacheForSingleFile.java [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.clearcache_forsinglefile_layout);        
+        mXWalkView = (XWalkView) findViewById(R.id.xwalkview);        
+    }
+
+    @Override
+    protected void onXWalkReady() {
+        StringBuffer mess = new StringBuffer();
+        mess.append("Test Purpose: \n\n")
+        .append("Verifies XWalkView can implement clearCacheForSingleFile API.\n\n")
```
Comment (Shao-Feng) [inline usecase/usecase-embedding-android-tests/embeddingapi/src/org/xwalk/embedded/api/sample/XWalkViewWithClearCacheForSingleFile.java]: can --> has?
or 

The method clearCaacheForSingleFile work in XWalkView

File: usecase/usecase-embedding-android-tests/embeddingapi-async/src/org/xwalk/embedded/api/asyncsample/XWalkViewWithClearCacheForSingleFileAsync.java (hunk 1 above)
Comment (Shao-Feng) [inline usecase/usecase-embedding-android-tests/embeddingapi-async/src/org/xwalk/embedded/api/asyncsample/XWalkViewWithClearCacheForSingleFileAsync.java]: Keep it shorter, you can see the warnning in Android Studio.

File: usecase/usecase-embedding-android-tests/embeddingapi-async/src/org/xwalk/embedded/api/asyncsample/XWalkViewWithClearCacheForSingleFileAsync.java (hunk 2 above)
Comment (Shao-Feng) [inline usecase/usecase-embedding-android-tests/embeddingapi-async/src/org/xwalk/embedded/api/asyncsample/XWalkViewWithClearCacheForSingleFileAsync.java]: use:
XWALK_CORE = "/app_xwalkcore/Default/Cache"
getCacheDir() + getPackageName+ XWALK_CORE

**Comment by Shao-Feng** [comment]
same comment for sync activity

**Comment by jacky-young** [comment]
@jennycao @Shao-Feng please help to review.

**Pull request merged**